app/src/main/java/com/example/goodnote/utils/Injectors.kt

```diff
@@ -1,14 +1,32 @@
 package com.example.goodnote.utils
 
 import android.content.Context
+import androidx.fragment.app.FragmentActivity
+import androidx.lifecycle.ViewModelProviders
 import com.example.goodnote.database.LocalDb
 import com.example.goodnote.database.repository.NoteRepo
 import com.example.goodnote.database.repository.NoteRepoImpl
+import com.example.goodnote.ui.viewModels.NoteViewModel
+import com.example.goodnote.ui.viewModels.NoteViewModelFactory
 
 object Injectors {
 
-    /*private?*/fun getNoteRepository(context: Context): NoteRepo {
+    private fun getNoteRepository(context: Context): NoteRepo {
         return NoteRepoImpl.getInstance(
-            LocalDb.getInstance(context.applicationContext).noteDao())
+            LocalDb.getInstance(context.applicationContext).noteDao()
```

Comment:
Think about moving NoteDao object creating to `val` property with lazy delegate. Creating db object is a heavy task, maybe it would be a good idea to cache its result :)

Reply:
Would the Injector object then need to get a context as parameter? Or should the dao creation be done like the repository/db instanaces (instance ?: getFromDb) ?

Reply:
like this `private var noteDao: NoteDao? = null
    private fun getNoteDao(context: Context): NoteDao {
        return noteDao ?: LocalDb.getInstance(context.applicationContext).noteDao().also { noteDao = it }
    }` ?

Reply:
That's it! :)

Reply:
yay :)